crates/core/src/dht/dht_store.rs

```diff
@@ -297,12 +302,37 @@ impl DhtStore {
         &mut self.actions
     }
 
-    pub(crate) fn next_queued_holding_workflow(&self) -> Option<&PendingValidation> {
-        self.queued_holding_workflows.front()
+    pub(crate) fn next_queued_holding_workflow(
+        &self,
+    ) -> Option<(&PendingValidation, Option<Duration>)> {
+        for (pending, maybe_delay) in self.queued_holding_workflows.iter() {
+            if let Some((time_of_dispatch, delay)) = maybe_delay {
+                let maybe_time_elapsed = time_of_dispatch.elapsed();
+                if let Ok(time_elapsed) = maybe_time_elapsed {
+                    if time_elapsed < *delay {
+                        continue;
+                    }
+                }
+                return Some((pending, Some(*delay)));
+            }
+            return Some((pending, None));
+        }
+        None
```

Comment:
Can we get away with a skip_while here, or will that make it more complex? It would be nice to see some comments here on what is happening either way

self.queued_holding_workflows
.iter()
.skip_while(|(pending,maybe_delay)|{/// comparison on time_elapsed})
.map(|(pending,maybe_delay)|(pending,maybe_delay.map(|f,s|Some(s)).unwrap_or(None)
next()